daemon-scheduler/pkg/engine/monitor.go

```diff
@@ -78,3 +110,22 @@ func (m monitor) runOnce() error {
 
 	return nil
 }
+
+func (m monitor) runPendingOnce() error {
+	environments, err := m.environment.ListEnvironments(m.ctx)
+	if err != nil {
+		return errors.New("Could not retrieve environments")
```

Comment:
Adding context here might be useful.. `Could not retrieve environments while running the pending monitor`.